engine/render/src/render/constant.cpp

```diff
@@ -201,11 +202,16 @@ void SetNamedConstantAtIndex(HNamedConstantBuffer buffer, dmhash_t name_hash, co
         constant.m_NameHash    = name_hash;
         constant.m_NumValues   = value_size;
         constant.m_ValueIndex  = values_index;
+        constant.m_Type        = constant_type;
         constants.Put(name_hash, constant);
 
         // Get the pointer
         c = constants.Get(name_hash);
     }
+    else if (c->m_NumValues > 0 && c->m_Type != constant_type)
+    {
+        return RESULT_TYPE_MISMATCH;
+    }
```

Comment:
This means that if you created a table with a different type that what you are trying to set currently, it will fail. This is the same in the case where a table has a single value, which maybe you want to be able to effectively change the type of the array? I.e, you create a table like so:

local a = render.render_constant()
a.arr = { vmath.vector4() }

now, you can't change a.arr to anything else.